src/uwtools/tests/drivers/test_ww3.py

```diff
@@ -0,0 +1,85 @@
+# pylint: disable=missing-function-docstring,protected-access,redefined-outer-name
+"""
+WW3 driver tests.
+"""
+from unittest.mock import DEFAULT as D
+from unittest.mock import patch
+
+import yaml
+from pytest import fixture
+
+from uwtools.drivers import ww3
+
+# Fixtures
+
+
+@fixture
+def config(tmp_path):
+    return {
+        "ww3": {
+            "namelist": {
+                "base_file": str(tmp_path / "ww3_shel.inp.IN"),
```

Comment:
This is not a namelist file and should not be used in conjunction with the `update_values` block. In so much as the unit tests serve as examples, we probably do not want to mix the approach here.

Reply:
Fair, though its also a valid input - should I put some logic that checks for this condition and bypasses the namelist validation, or at least accepts another input? Or perhaps this should be hard-coded and error on other inputs?

Reply:
@christinaholtNOAA I'm not sure I understand your concern on this one. What is the "mix[ed] approach"? The output _is_ a namelist file, so I would think that `namelist:`, at least, is appropriate. Is it `base_file` and `update_values` that you're concerned about? My thinking in suggesting this to @WeirAE was that:

1. If we are later able to update our Fortran namelist handling to deal with the array and user-defined-type syntax used in the WW3 namelists, the change could be _internal only_ and not affect the UW YAML language used here. That is, users would not have to change their configs -- they probably wouldn't even need to change their input namelist (template) files, as `f90nml` could just overwrite the Jinja2 expressions with real namelist values. If we use some other terminology here, then the UW YAML would have to change, too.
2. We also use `base_file` and `update_values` for INI, SH, and YAML configs -- they're not specific to NML configs.
3. And, anyway, aren't we updating a base file with some values? It seems like appropriate -- not to mention consistent -- terminology.

But I might have totally misunderstood your intent here. Sorry if so.

Reply:
Aha, there is a way to link comments! To step in and sort a little, this was @christinaholtNOAA 's clarification on file limits [here](https://github.com/ufs-community/uwtools/pull/496#discussion_r1617908543) where the [input script](https://github.com/NOAA-EMC/WW3/blob/d9b3172f4197c65d471662c6952a668152d71230/model/src/wav_shel_inp.F90) used by UFS checks for either [ww3_shel.inp](https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/parm/ww3_shel.inp.IN) or [ww3_shel.nml](https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/parm/ww3_shel.nml.IN). The .inp file is parsed by the input script, which we're not overriding, so I think it might be ok to accept but bypass it, and change to just the nml for the tests? It looks like atparse to me so we _could_ even handle it properly

Reply:
I think Brian is correct. WW3 accepts either an ascii `ww3_shel.inp` file as its source of configuration information **or** it accepts a legit fortran namelist named `ww3_shel.nml`. We should not be suggesting that users can put a `ww3_shel.inp.IN` template input file in here since that most likely corresponds to the ascii version of the file. Even though the driver will fill it in as a template just fine, when it's placed in the run directory as `ww3_shel.nml` (hardcoded in the driver), it won't be formatted as expected and the program will (hopefully) die.

Reply:
Yes! I understand now. It was just a concern about the example filename here, `ww3_shel.inp.IN`, being liable to mislead users because _that_ file is not a namelist file. I thought it was a deeper concern about how we're provisioning the actual namelist file. Please let me know if I'm still off the mark.

Reply:
Nope. I think we're on the same page now.

Reply:
Offline discussion: a clear breaking change is preferable, so we should define it as 'template_file' and 'template_values'